sources/xfscli/src/bin/xfs_admin.rs

```diff
@@ -0,0 +1,211 @@
+use std::ops::Add;
+use std::process::exit;
+use std::process::Command;
+use std::str;
+
+use argh::FromArgs;
+
+/// change parameters of an XFS filesystem
+#[derive(FromArgs)]
+struct Args {
+    /// enables unwritten extent support on a filesystem that is disabled
+    #[argh(switch, short = 'e')]
+    extflg: bool,
+    /// specifies that the filesystem image to be processed is stored in a regular file at device
+    #[argh(switch, short = 'f')]
+    fs_image: bool,
+    /// enables version 2 log format
+    #[argh(switch, short = 'j')]
+    log2: bool,
+    /// prints the current filesystem label
+    #[argh(switch, short = 'l')]
+    fs_label: bool,
+    /// enable 32bit project identifier support
+    #[argh(switch, short = 'p')]
+    projid32bit: bool,
+    /// prints the current filesystem UUID
+    #[argh(switch, short = 'u')]
+    fs_uuid: bool,
+    /// enable (1) or disable (0) lazy-counters in the filesystem
+    #[argh(option, short = 'c')]
+    lazy_counters: Option<usize>,
+    /// set the filesystem label to label
+    #[argh(option, short = 'L')]
+    label: Option<String>,
+    /// add or remove features on an existing V5 filesystem
+    #[argh(option, short = 'O')]
+    v5_feature: Option<String>,
+    /// set the UUID of the filesystem to uuid
+    #[argh(option, short = 'U')]
+    uuid: Option<String>,
+    /// specifies the device special file where the filesystem's realtime section resides
+    #[argh(option, short = 'r')]
+    rtdev: Option<String>,
+    /// block-device
+    #[argh(positional)]
+    device: Option<String>,
+    /// filesystem's external log location
```

Comment:
nit: Its common practice to separate positional from options so that the reader can find those easily. Please move this to the bottom and separate with a space.